### ISSUE TITLE
WebFlux AbstractView does not allow null model values anymore

### ISSUE THREAD

**Issue by bclozel** [spring-projects/spring-framework#23038]
As of 172afb510a7e8e0daca10753fe965b0dfb53f6f7, the `AbstractView` implementation switched from a `LinkedHashMap` to a `ConcurrentHashMap` for storing model attributes. The latter does not allow `null` values and will throw a `NullPointerException` if the provided model has a `null` value associated with any key.

This is currently breaking the Spring Boot build as we're adding null values to the model in the error handling support.

Should we try and relax that requirement or just enforce it by filtering null values before copying them in the `attributes` map?

**Comment by rstoyanchev** [comment]
WebFlux uses `ConcurrentModel` and copying that into a `LinkedHashMap` in `AbstractView` seemed to promise preserving an order that wasn't there in the first place. The impact on null values was not intended. 

Currently we also [remove async attributes](https://github.com/spring-projects/spring-framework/blob/4b8ce04bdc45c0e523a2b2e140cdc6b4f8db858a/spring-webflux/src/main/java/org/springframework/web/reactive/result/view/AbstractView.java#L267) that complete without a value. 

I suppose we should make these two consistent. Any preference which way? From a Boot perspective is it okay to render without the "message" attribute if there is no message?

**Comment by bclozel** [comment]
To me, a good workaround would be not to insert null values at all in the `ConcurrentHashMap`.
I assume that in a Controller handler, something might do something like:

```java
User user = fetchUser("brian"); // user might be null
model.addAttribute("user", user);
```

As a developer, I'd rather have the key missing from the map than an `NullPointerException` being raised at runtime. Now I don't know if we want to achieve full consistency with how things are supported in Spring MVC (there, I believe we allow null values in the map). In this case, removing null values would be also consistent with removing async attributes that complete without a value.

**Comment by rstoyanchev** [comment]
Just noticing #21678, so in the above scenario the attribute would be removed, which also points in the direction of filtering them out in AbstractView.

**Comment by philwebb** [comment]
This is also in 5.1.x so can the fix also be backported please

**Comment by rstoyanchev** [comment]
Good point @philwebb although no need to backport. I'll fix it in 5.1.x and forward merge.
